tpot/base.py

```diff
@@ -238,6 +240,7 @@ def __init__(self, generations=100, population_size=100, offspring_size=None,
         self.early_stop = early_stop
         self._last_optimized_pareto_front = None
         self._last_optimized_pareto_front_n_gens = 0
+        self.memory = None
```

Comment:
Let's make `memory` another TPOT parameter, with the following options:

* `None` (default): No memory
* `'auto'`: Memory with a temp directory (TPOT creates the temp directory and cleans it up upon shutdown)
* directory path string: Memory with the provided directory (TPOT does NOT clean the directory up upon shutdown)
* `Memory` object: Memory with provided `Memory` object (TPOT does NOT clean the directory up upon shutdown)

Reply:
Make sense, will do

Reply:
In the docs for these options, we should clarify that we allow the user to specify a custom directory path or `Memory` object in case they want to re-use the cache in future TPOT runs (or a `warm_start` run). However, they shouldn't expect the cache to be meaningfully interpretable in any way.